src/atc/cache/CachedLoaderParameters.py

```diff
@@ -0,0 +1,29 @@
+# Defines input parameters for a cached Loader
+
+from typing import List
+
+
+class CachedLoaderParameters:
+    def __init__(
+        self,
+        cache_table_name: str,
+        key_cols: List[str],
+        cache_id_cols: List[str] = None,
+    ):
+        """
+        Args:
+            cache_table_name: The table that holds the cache
+            key_cols: the set of columns that form the primary key for a row
+            cache_id_cols: These columns, added by the write operation, will be saved
+                in the cache to identify e.g. the written batch.
+        """
+
+        self.cache_id_cols = cache_id_cols or []
+        self.cache_table_name = cache_table_name
```

Comment:
if not cache_table_name:
            raise ValueError("cache_table_namecolumns must be provided")

Reply:
this is alreadu checked as part of the `validate` method